src/Impostor.Server/Net/Inner/Objects/InnerPlayerControl.Api.cs

```diff
@@ -103,5 +106,19 @@ public async ValueTask SetMurderedAsync()
             writer.Write((byte)NetId);
             await _game.FinishRpcAsync(writer);
         }
+
+        public async ValueTask SetMurderedByAsync(IClientPlayer impostor)
+        {
+            var impostorNetId = impostor.Character.NetId;
```

Comment:
Check if the `impostor` is actually an impostor, check if they're not dead, and also check if the target player is not dead.